src/App.tsx

```diff
@@ -15,36 +17,100 @@
   'Garlic',
 ];
 
+let goods = [...goodsFromServer];
+let isReverseCLicked = false;
+
 export const App: React.FC = () => {
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortGoods, setSortGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function setSort() {
+    if (sortGoods === ALPHABETICALLY) {
+      goods = [...goodsFromServer].sort((goodA, goodB) =>
+        goodA.localeCompare(goodB),
+      );
+    }
+
+    if (sortGoods === BY_LENGTH) {
+      goods = [...goodsFromServer].sort(
+        (goodA, goodB) => goodA.length - goodB.length,
+      );
+    }
+
+    if (reverse) {
+      goods.reverse();
+    }
+  }
```

Comment:
[CODE STYLE] Function's name should start with a verb. Also, it's better to avoid side effects in your functions. Consider refactoring `setSort` to return a new sorted array instead of modifying the `goods` array directly.